### DISCUSSION TITLE
part-ia/2012-29/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#426] [Question comments]
Part IA 2012 Numbers and Sets: Paper 4, Section II, D 

https://questions.tripos.org/part-ia/2012-29/

**Comment by hennalVEVO** [comment]
<details>
<summary>Solution outline</summary>

Suppose there were such an injection, this would imply the existence of a surjective $f : \mathbb{R} \to \mathcal{P}(\mathbb{R})$ and we now prove this as standard from lecture notes. For the next part, we use scaled and translated $\operatorname{arctan}$ to determine a suitable injection $g : \mathbb{R}^2 \to (0,1)^2$. We interlace digits in the obvious way to get injection $h : (0,1)^2 \to (0,1)$ and since $(0,1) \subset \mathbb{R}$, we have our required injection $h \circ g$ .


For the last part, notice that for each $f(x) \neq x$ we have two unique real values $x_k$ and $f(x_k)$ defining this point. We can express each $f$ as the sequence $x_1, f(x_1), \dots, x_n, f(x_n), 0, 0, \dots$, taking $x_k < x_{k+1}$ for simplicity's sake. We can define $g' : \mathbb{R}^{\mathbb{N}} \to (0,1)^{\mathbb{N}}$ by $g'((a_n)^{\infty}_{n=1}) = (g(a_n))^{\infty}_{n=1}$ where $g$ is the same modified $\operatorname{arctan}$ as above. Now we interlace digits appropriately to get injective $h' : (0,1)^{\mathbb{N}} \to (0,1)$. We do this in a pyramid fashion, $0.a_{1}a_{2}b_{1}a_{3}b_{2}c_{1}\dots$ where $a, b, c \in (0,1)$ are the first $3$ elements of our sequence and $a_n$ is the $n^{\text{th}}$ digit of $a$. Since $(0,1) \subset \mathbb{R}$ and $X \subset \mathbb{R}^{\mathbb{N}}$, we have our required injection $h' \circ g'$.
</details>
